### ISSUE TITLE
UnexpectedEof when using for_each_chunk

### ISSUE THREAD

**Issue by vektrixdev** [owengage/fastnbt#21]
The following code:
```rust
use fastanvil::Region;
use std::fs::File;

fn main() {
    let file =  File::open("./region/r.0.0.mca").unwrap();
    let mut r = Region::new(file);
    r.for_each_chunk(|x, z, data| {
        println!("{}, {}", x,  z);
    }).expect("hhh");
}
```
gives this error and no other output
```
thread 'main' panicked at 'hhh: IO(Custom { kind: UnexpectedEof, error: "failed to fill whole buffer" })', src/main.rs:9:8
```
I've tried with 1.15 and 1.16 region files and had no luck.

**Comment by owengage** [comment]
Hey there. Your code looks fine so looks like a bug. Could you potentially upload that particularly Region file somewhere so I can try to reproduce it?

**Comment by vektrixdev** [comment]
[region.zip](https://github.com/owengage/fastnbt/files/5609160/region.zip)

**Comment by owengage** [comment]
Awesome. I'll have a look when I can. :)

**Comment by owengage** [comment]
I've released fastanvil 0.14.1 that should fix this.

This seems to be due to one of the chunks in the region not being padded to a multiple of 4096 bytes as the (wiki)[https://minecraft.gamepedia.com/Region_file_format] claims it has to be. The chunk metadata includes to proper unpadded length, so I now use that instead.
